### ISSUE TITLE
Space in ridge plot

### ISSUE THREAD

**Issue by nilseling** [dtm2451/dittoSeq#63]
Hi Daniel,

first of all: thanks for this great package! It really saves so much time plotting things!
I'm a huge fan of the `multi_dittoPlot` function but always notice a gap at the upper part of the y-axis (see attached).
I generated the plot with the following call:

```
multi_dittoPlot(sce, vars = rownames(sce)[1:2], group.by = "ImageNumber", plots = "ridgeplot", assay = "exprs") 
```

Here, `exprs` contains arcsinh-transformed counts (not working with scRNAseq data anymore ;) ) and `ImageNumber` can be seen as a patient identifier.
Do you have a suggestion how to get rid of this gap?

![image](https://user-images.githubusercontent.com/11314064/98117352-e7ab4680-1ea9-11eb-9af9-2792ea4b6096.png)

**Comment by dtm2451** [comment]
Oof.  I haven't built in anything for tweaking that, but it looks like I did shrink the output for ridgeplots with `scale_y_discrete(expand = expansion(mult=c(0, 0.68)))`... and I'm not sure why anymore.

If you install from the `ridge-adjust` branch,
`BiocManager::install("dtm2451/dittoSeq@ridge-adjust")` and try again does it look better?

**Comment by nilseling** [comment]
Sorry for the late reply - I must have missed the notification.
Your change did the trick:

**Comment by dtm2451** [comment]
Awesome!

I'm definitely keeping the adjustment in some way because I remember why I had it now.  Look at that top ridge, for imagename 43... it's cut off.

I'll probably add back the code that I'd removed here, but after playing around with its default and after adding in some input, maybe `ridgeplot.y.shrinkage`, to make it easy for a user to adjust.

Otherwise well... the way you would manually adjust this particular bit with `multi_dittoPlot` would be UGLY. For single plots, it's easier, but the multis... It'd require something like the below this which I do not want to force on users:

```
lapply(
    multi_dittoPlot(..., list.out = TRUE),
    function(x) {
        x + <theme_adjustment>
    })
```

To be continued =)

**Comment by nilseling** [comment]
Yes, I noticed the ridge being cut off. And yes, having some sensible default e.g. `ridgeplot.y.shrinkage = 0.95` or `ridgeplot.y.range = c(0, 0.95)` would make sense. I guess it always depends on the layout of the graphics device and the shape of the distribution how bad the cut is. 

I checked and adding `... + coord_cartesian(clip = "off")` to each individual plot got rid of the clipping. But it also overwrote some internal coordinate system that you are using - so not sure if that's a viable option.

**Comment by dtm2451** [comment]
It seems to me that no one value will suffice for the `scale_y_discrete(expand = expansion(mult=c(0, some_value)))` code because the needed value should be large for fewer groupings, but small for many groupings. Also seems that ggplot is pretty good at adjusting the plotting extents to fit the data on its own, with the top-ridge always extending right to the top of the plotting extent. So... simplest and best option probably is the `coord_cartesian(clip="off")` option you propose.  If only that didn't muck up the `min`imum & `max`imum extent adjustment of the x-axis whic is what I currently have going into `coord_cartesian()`.

So, sorry for the delay.
On the + coord_cartesian(clip = "off") option: As you noticed, this does overwrite a previous usage of the function by dittoSeq. It's used for trimming the x-axis to the `min` & `max` input values.  But really, if you are going to turn clipping off, you would not want to be zoomed into your data in any way, because then you might have non-clipped data overlapping with the axes-labels.  Overall, it's not aa bad, simple option here that would always work.

**Comment by dtm2451** [comment]
On the `coord_cartesian(clip = "off")` option: As you noticed, this does overwrite a previous usage of the function by dittoSeq. It's used for trimming the x-axis to the `min` & `max` input values. In truth, if you are going to turn clipping off, you would not want to be zoomed into your data in any way, because then you might have non-clipped data overlapping with the axes-labels. Overall, it's not a bad, simple option here that should always work for the current goal of fixing the spacing issue.  BUT it would be hard to make this option work while still keeping the `min` `max` input functionality, and I don't want to remove that.

So what I've done:
- Made a `ridgeplot.ymax.expansion` input that is NA by default, but allows user-correction when needed.
- This variable gets used within the `scale_y_discrete(expand = expansion(mult=c(0, HERE)))` axis adjustment method.
- When left as NA, the value is filled in based on linear interpolation between values that I identified as "generally good" for plots with 3, 12, and 34 ridges in them.

I don't love this solution, but I think it gets the job done well enough.

**Comment by dtm2451** [comment]
Let me know if this looks good with your data??

**Comment by nilseling** [comment]
Hey, thanks for your work on this!
I installed the latest version from the `ridge-adjust` branch and the plot now looks like this:

![image](https://user-images.githubusercontent.com/11314064/99640376-164c2400-2a49-11eb-893f-a752fd6d4635.png)

So this is nice! This approach is definitely good enough and I'll close this issue.
Thanks again!

**Comment by dtm2451** [comment]
Just pushed this "bug fix" to both release (3.12.1) and devel (3.13.1).  Couple days and it'll be worked into the normal `BiocManager::install("dittoSeq")` =)
